overlord/ifacestate/schemas/schemas.go

```diff
@@ -0,0 +1,44 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package schemas holds structs for reading and writing interface-related data.
+package schemas
+
+import "github.com/snapcore/snapd/snap"
+
+// Connection holds properties of an interface connection.
+type Connection struct {
```

Comment:
I think ConnState or ConnectionState matches more our conventions (see SnapState or overlord/auth.*State). Also we have already too many things called Connection and this doesn't have the property of in itself containing the plug and slot as those others